Python/compile.c

```diff
@@ -558,8 +562,8 @@ compiler_unit_free(struct compiler_unit *u)
     PyMem_Free(u);
 }
 
-static struct compiler_unit *
-get_class_compiler_unit(struct compiler *c)
+static int
+compiler_add_static_attribute_to_class(struct compiler *c, PyObject *attr)
```

Comment:
Notes for me more than for you: This function was only called from one place, and the change moves more logic from the call site into the function.

Reply:
Yes, and this function is now a "compiler function" which can access compiler internals while the calling function is a codegen function which can work with an opaque `struct compiler*`.